src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -587,20 +617,41 @@
                 // see if we can decode, decrypt and deserialize the users token
                 // into a login session structure
                 let session = self.session_cache.decode(token, &self.session_key)?;
-
                 let status = session.status();
 
-                let mut new_auth = None;
-                if status != SessionStatus::Active {
-                    new_auth = self.try_refresh_token(&session);
-                    if new_auth.is_none() && status == SessionStatus::Expired {
-                        return Err(Error::ApiInvalidCredentials(
-                            "Session expired, please login again".to_string(),
-                        ));
-                    }
+                // Token found in cache and active; all good, do an early return
+                if status == SessionStatus::Active {
+                    return Ok(Some(ActorDef::user(session.id, session.attributes, None)));
                 }
 
-                Ok(Some(ActorDef::user(session.id, session.attributes, new_auth)))
+                let new_auth = match self.try_refresh_token(&session) {
+                    Ok(auth) => auth,
+                    Err(e) => {
+                        warn!("RFC-6749 5.2 Error Response returned: {:?}", e);
+                        match e {
```

Comment:
These match handlers discard the original error details, is there nothing useful from the original error to return to the caller?

Reply:
Probably yes, I assumed this would be embedded in a more debug level message about the context